drivers/pinctrl/pinctrl_nrf.c

```diff
@@ -352,6 +355,20 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
 		if (psel != PSEL_DISCONNECTED) {
 			uint32_t pin = psel;
 
+#ifdef CONFIG_SOC_NRF54H20_GPD
+			if (NRF_GET_GPD_FAST_ACTIVE1(pins[i]) == 1U) {
+				int ret;
+				NRF_GPIO_Type *port = nrf_gpio_pin_port_decode(&pin);
```

Comment:
This changes `pin` to a relative number within the `port`, but further `nrf_gpio_*` calls require `pin` in its absolute form. You need to use a local variable for this conversion.

Reply:
thanks good catch